src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -75,54 +76,91 @@ protected virtual IEnumerable<FieldType> ProvideFields()
+                        queryArguments.Add(queryArgument);
+                    }
+                    expression ??= AutoRegisteringHelper.GetParameterExpression(
+                        parameterInfo.ParameterType,
+                        queryArgument ?? throw new InvalidOperationException("Invalid response from ConstructQueryArgument: queryArgument and expression cannot both be null"));
+                    expressions.Add(expression);
                 }
-                expression ??= AutoRegisteringHelper.GetParameterExpression(
-                    parameterInfo.ParameterType,
-                    queryArgument ?? throw new InvalidOperationException("Invalid response from ConstructQueryArgument: queryArgument and expression cannot both be null"));
-                expressions.Add(expression);
+                var resolver = new MethodResolver(methodInfo, BuildMemberInstanceExpression(memberInfo), expressions);
+                fieldType.Arguments = queryArguments;
+                fieldType.Resolver = resolver;
+            }
+            else if (memberInfo is FieldInfo fieldInfo)
+            {
+                var sourceExpression = BuildMemberInstanceExpression(memberInfo);
+                var param = sourceExpression.Parameters[0];
+                var body = Expression.Convert(
+                    Expression.MakeMemberAccess(sourceExpression.Body, fieldInfo),
+                    typeof(object));
+                var lambda = Expression.Lambda<Func<IResolveFieldContext, object?>>(body, param);
+                var func = lambda.Compile();
+                fieldType.Arguments = null;
+                fieldType.Resolver = new FuncFieldResolver<object>(func);
+            }
+            else if (memberInfo == null)
+            {
+                throw new ArgumentNullException(nameof(memberInfo));
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException(nameof(memberInfo), "Member must be a field, property or method.");
             }
-            var resolver = new MethodResolver(methodInfo, expressions);
-            return (queryArguments, resolver);
         }
 
+        /// <summary>
+        /// Returns a lambda expression that will be used by the field resolver to access the member.
+        /// <br/><br/>
+        /// Typically this is a lambda expression of type <see cref="Func{T, TResult}">Func</see>&lt;<see cref="IResolveFieldContext"/>, <typeparamref name="TSourceType"/>&gt;.
+        /// <br/><br/>
+        /// By default this returns the <see cref="IResolveFieldContext.Source"/> property.
+        /// </summary>
+        /// <param name="memberInfo">The member being called or accessed.</param>
+        protected virtual LambdaExpression BuildMemberInstanceExpression(MemberInfo memberInfo)
+            => _sourceExpression;
+
+        private static readonly Expression<Func<IResolveFieldContext, TSourceType>> _sourceExpression = context => (TSourceType)context.Source!;
```

Comment:
I believe NullReferenceException is the proper exception type in this case.

```suggestion
        private static readonly Expression<Func<IResolveFieldContext, TSourceType>> _sourceExpression
            = context => (TSourceType)context.Source! ?? ThrowSourceNullException();

        private static TSourceType ThrowSourceNullException()
        {
            throw new NullReferenceException($"IResolveFieldContext.Source is null; please use static methods when using an AutoRegisteringGraphType as a root graph type.");
        }
```

Note: throw is not permitted inside an expression

Reply:
I updated comment. Commit if OK.